torch/fx/subgraph_rewriter.py

```diff
@@ -27,11 +29,15 @@ def matches_subgraph_from_anchor(self, anchor : Node) -> bool:
         Pattern matching is done by recursively comparing the pattern
         node's use-def relationships against the graph node's.
         """
+        if anchor.op == "placeholder":
+            return False
         self.nodes_map = {}
-        return self._match_nodes(self.pattern_anchor, anchor)
+        return (self._match_nodes(self.pattern_anchor, anchor)
+                and len(self.nodes_map) == len(self.pattern.nodes))
```

Comment:
What situation is this fixing? It's not immediately apparent to me what this condition ensures